packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1928,6 +1928,41 @@ export class MergeTree {
 			localSeq,
 			segmentGroup: undefined,
 		};
+		const normalizedStartPos = startPos === "start" || startPos === undefined ? 0 : startPos;
```

Comment:
Consider the empty obliteration cases too. What's preventing `startPos === 'end'`?

Reply:
I added the following function on my branch for this purpose:
```typescript
/**
 * Returns the given place in InteriorSequencePlace form.
 */
export function normalizePlace(place: SequencePlace): InteriorSequencePlace {
	if (typeof place === "number") {
		return { pos: place, side: Side.Before };
	}
	if (place === "start") {
		return { pos: -1, side: Side.After };
	}
	if (place === "end") {
		return { pos: -1, side: Side.Before };
	}
	return place;
}
```

Reply:
With an `InteriorSequencePlace` you can do the following:
```typescript
const startSegment = this.getContainingSegment(startPlace.pos, refSeq, clientId).segment;
obliterate.start = this.createLocalReferencePosition(
	startSegment,
	startPlace.side === Side.Before ? 0 : startSegment?.cachedLength - 1,
	ReferenceType.StayOnRemove | ReferenceType.RangeBegin,
	{ obliterate },
);
```